lib/v2/screens/receive_screens/receive_enter_data/receive_seeds_screen.dart

```diff
@@ -1,69 +1,107 @@
+              listenWhen: (_, current) => current.pageCommand != null,
+              listener: (context, state) {
+                if (state.pageCommand is NavigateToReceiveDetails) {
+                  //Navigate to Receive Detail Page
+                }
+                if (state.pageCommand is ShowTransactionFailSnackBar) {
+                  ScaffoldMessenger.of(context).showSnackBar(
+                    SnackBarInfo(title: 'Receive creation failed, try again.', context: context),
+                  );
+                }
+              },
+              builder: (context, ReceiveEnterDataState state) {
+                switch (state.pageState) {
+                  case PageState.initial:
+                    return const SizedBox.shrink();
+                  case PageState.loading:
+                    return const FullPageLoadingIndicator();
+                  case PageState.failure:
+                    return const FullPageErrorIndicator();
+                  case PageState.success:
+                    return Stack(
                       children: [
-                        TextFormFieldLight(
-                          labelText: "Description",
-                          hintText: "Enter Product Details",
-                          maxLength: blockChainMaxChars,
-                          onChanged: (String value) {
-                            // BlocProvider.of<ReceiveEnterDataPageBloc>(context)
-                            //     .add(OnDescriptionChange(descriptionChanged: value));
-                          },
-                        ),
-                        const SizedBox(height: 16),
-                        const BalanceRow(
-                          label: "Available Balance",
-                          fiatAmount: "TODO",
-                          seedsAmount: "TODO",
+                        SingleChildScrollView(
+                          child: Column(
+                            crossAxisAlignment: CrossAxisAlignment.start,
+                            children: [
+                              const SizedBox(height: 100),
+                              AmountEntryWidget(
+                                onValueChange: (value) {
+                                  BlocProvider.of<ReceiveEnterDataBloc>(context)
+                                      .add(OnAmountChange(amountChanged: value));
+                                },
+                                autoFocus: state.isAutoFocus,
+                              ),
+                              const SizedBox(height: 36),
+                              Padding(
+                                padding: const EdgeInsets.symmetric(horizontal: horizontalEdgePadding),
+                                child: Column(
+                                  children: [
+                                    TextFormFieldLight(
+                                      labelText: "Description",
+                                      hintText: "Enter Product Details",
```

Comment:
enter product details?

Reply:
thast how we have it on figma. should I change it for " Add a description" ?

Reply:
What if you want to receive a payment, not related to a product? I dont think `product` should be used here 

Change it to `Enter a description.`

Reply:
hintText was changed to "Enter a Description"